src/printer.rs

```diff
@@ -239,6 +240,34 @@ impl<'a> Printer<'a> {
                     );
                 });
             }
+            SerializationFormat::Gitlab => {
+                // Generate JSON with errors in GitLab CI format
+                // https://docs.gitlab.com/ee/ci/testing/code_quality.html#implementing-a-custom-tool
+                println!(
+                    "{}",
+                    serde_json::to_string_pretty(
+                        &diagnostics
+                            .messages
+                            .iter()
+                            .map(|message| {
+                                json!({
+                                    "description": format!("({}) {}", message.kind.code(), message.kind.body()),
+                                    "severity": "major",
+                                    "fingerprint": message.kind.code(),
+                                    "location": {
+                                        "path": relativize_path(Path::new(&message.filename)),
+                                        "lines": {
+                                            "begin": message.location.row(),
+                                            "end": message.end_location.row()
+                                        }
+                                    }
+                                })
```

Comment:
Not sure if we should add `struct`'s to structure the JSON. It may be better to use struct's even if we will use it for just this one use-case. (I have recently started learning Rust)

Reply:
I think this is fine for now, since it's treated equivalently to the GitHub approach and colocated with that formatting anyway.